R/target_sda.R

```diff
@@ -294,7 +301,7 @@ compute_loanbook_targets <- function(data,
                                      scenario_with_p,
                                      ...) {
   data %>%
-    inner_join(scenario_with_p, by = c("year", "sector")) %>%
+    full_join(scenario_with_p, by = c("year", "sector")) %>%
```

Comment:
Why this change? Just curious if it relates to some issue other than interpolation.

Reply:
this is the crucial change for:
> target_sda() will now calculate the convergence target to the final year present in scenario, regardless of how many years we have data for in ald. (Compared to previously, when the convergence target was erroneously calculated to the last year present in all).

`inner_join` would only keep track of the scenario values if there was a corresponding year in `ald`, but we need to know scenario values for any year.

Reply:
`full_join` allows us to retain all info from both datasets (which is desirable)